src/main/java/it/pagopa/pn/paperchannel/middleware/queue/consumer/handler/RECAG011BMessageHandler.java

```diff
@@ -72,11 +72,11 @@ public Mono<Void> handleMessage(PnDeliveryRequest entity, PaperProgressStatusEve
                 .flatMap(pnEventDemats ->  eventMetaDAO.getDeliveryEventMeta(metadataRequestIdFilter, META_SORT_KEY_FILTER ))
                 .doOnNext(pnEventMeta -> log.info("[{}] PnEventMeta found: {}", paperRequest.getRequestId(), pnEventMeta))
                 .map(pnEventMetaRECAG012 -> createMETAForPNAG012Event(paperRequest, pnEventMetaRECAG012, ttlDaysMeta))
-                .doOnNext(pnEventMeta -> pnLogAudit.addsBeforeReceive(entity.getIun(), String.format("prepare requestId = %s Response from external-channel", entity.getRequestId())))
+                .doOnNext(pnEventMetaPNAG012 -> editPnDeliveryRequestAndPaperRequestForPNAG012(entity, paperRequest, pnEventMetaPNAG012.getStatusDateTime()))
+                .doOnNext(pnEventMetaPNAG012 -> pnLogAudit.addsBeforeReceive(entity.getIun(), String.format("prepare requestId = %s Response from external-channel", entity.getRequestId())))
                 .flatMap(eventMetaDAO::createOrUpdate)
-                .doOnNext(pnEventMeta -> logSuccessAuditLog(paperRequest, entity, pnLogAudit))
-                .doOnNext(pnEventMeta -> editPnDeliveryRequestForPNAG012(entity))
-                .doOnNext(pnEventMeta -> log.info("[{}] Sending PNAG012 to delivery push: {}", paperRequest.getRequestId(), entity))
+                .doOnNext(pnEventMetaPNAG012 -> logSuccessAuditLog(paperRequest, entity, pnLogAudit))
+                .doOnNext(pnEventMetaPNAG012 -> log.info("[{}] Sending PNAG012 to delivery push: {}", paperRequest.getRequestId(), entity))
```

Comment:
un dubbio, qui è ok salvare auditlog di success prima di inviare a delivery push.
inoltre, se qualcosa va male, è gestito il log di failure?

Reply:
L'audit log è sulla generazione dell'evento, non sull'inoltro a DeliveryPush, questo consente di avere la stessa gestione del caso in cui l'evento sia ricevuto dal consolidatore. Non ci sarebbe bisogno di fare before ed after ma al momento è necessario per come sono pensati gli audit. La gestione del fallimento sull'inoltro a DP dovrebbe essere gestito nel punto in cui paper-channel fa la send su coda